packages/grafana-ui/src/components/Table/TableNG/Cells/PillCell.tsx

```diff
@@ -0,0 +1,98 @@
+import { classicColors } from '@grafana/data';
+
+import { PillCellProps } from '../types';
+
+function normalizePillValues(value: unknown): string[] {
+  if (Array.isArray(value)) {
+    return value.map(String).map(stripQuotes);
+  }
+  if (typeof value === 'string') {
+    // Try JSON parse
+    try {
+      const arr = JSON.parse(value);
+      if (Array.isArray(arr)) {
+        return arr.map(String).map(stripQuotes);
+      }
+    } catch (e) {
+      // Not JSON, continue
+    }
+    // Split by comma, but keep quoted substrings together
+    // e.g. 'foo,"Tim Levett",bar' => ['foo', 'Tim Levett', 'bar']
+    const regex = /(?:"([^"]*)")|(?:'([^']*)')|([^,]+)/g;
```

Comment:
i do love me some regexp, but this might be a bit rich even for my taste :sweat_smile: